src/main/java/core/basesyntax/FigureSuplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSuplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int RANDOM_LIMIT = 47;
+    private static final int DEFOLT_RADIUS = 10;
+    private static final String DEFOLT_COLOR = "white";
+    private Random random;
+    private ColorSuplier colorSuplier;
+
+    public FigureSuplier(Random random, ColorSuplier colorSuplier) {
+        this.random = random;
+        this.colorSuplier = colorSuplier;
+    }
```

Comment:
It's better to move the initialization of this variables to their declaration (line 10-11)